st2client/tests/unit/test_execution_tail_command.py

```diff
@@ -0,0 +1,310 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+import json
+
+from tests import base
+from tests.base import BaseCLITestCase
+
+from st2client.utils import httpclient
+from st2client.commands.action import LIVEACTION_STATUS_SUCCEEDED
+from st2client.commands.action import LIVEACTION_STATUS_FAILED
+from st2client.commands.action import LIVEACTION_STATUS_RUNNING
+from st2client.shell import Shell
+
+__all__ = [
+    'ActionExecutionTailCommandTestCase'
+]
+
+# Mock objects
+MOCK_LIVEACTION_1 = {
+    'id': 'idfoo1',
+    'status': LIVEACTION_STATUS_SUCCEEDED
+}
+
+MOCK_LIVEACTION_2 = {
+    'id': 'idfoo2',
+    'status': LIVEACTION_STATUS_FAILED
+}
+
+MOCK_LIVEACTION_3_RUNNING = {
+    'id': 'idfoo3',
+    'status': LIVEACTION_STATUS_RUNNING
+}
+
+# Mock liveaction objects for ActionChain workflow
+MOCK_LIVEACTION_3_CHILD_1_RUNNING = {
+    'id': 'idchild1',
+    'context': {
+        'parent': {
+            'execution_id': 'idfoo3'
+        },
+        'chain': {
+            'name': 'task_1'
+        }
+    },
+    'status': LIVEACTION_STATUS_RUNNING
+}
+
+MOCK_LIVEACTION_3_CHILD_1_SUCCEEDED = {
+    'id': 'idchild1',
+    'context': {
+        'parent': {
+            'execution_id': 'idfoo3'
+        },
+        'chain': {
+            'name': 'task_1'
+        }
+    },
+    'status': LIVEACTION_STATUS_SUCCEEDED
+}
+
+MOCK_LIVEACTION_3_CHILD_1_OUTPUT_1 = {
+    'execution_id': 'idchild1',
+    'timestamp': '1505732598',
+    'output_type': 'stdout',
+    'data': 'line 4\n'
+}
+
+MOCK_LIVEACTION_3_CHILD_1_OUTPUT_2 = {
+    'execution_id': 'idchild1',
+    'timestamp': '1505732598',
+    'output_type': 'stderr',
+    'data': 'line 5\n'
+}
+
+MOCK_LIVEACTION_3_CHILD_2_RUNNING = {
+    'id': 'idchild2',
+    'context': {
+        'parent': {
+            'execution_id': 'idfoo3'
+        },
+        'chain': {
+            'name': 'task_2'
+        }
+    },
+    'status': LIVEACTION_STATUS_RUNNING
+}
+
+MOCK_LIVEACTION_3_CHILD_2_FAILED = {
+    'id': 'idchild2',
+    'context': {
+        'parent': {
+            'execution_id': 'idfoo3'
+        },
+        'chain': {
+            'name': 'task_2'
+        }
+    },
+    'status': LIVEACTION_STATUS_FAILED
+}
+
+MOCK_LIVEACTION_3_CHILD_2_OUTPUT_1 = {
+    'execution_id': 'idchild2',
+    'timestamp': '1505732598',
+    'output_type': 'stdout',
+    'data': 'line 100\n'
+}
+
+MOCK_LIVEACTION_3_SUCCEDED = {
+    'id': 'idfoo3',
+    'status': LIVEACTION_STATUS_SUCCEEDED
+}
+
+# Mock objects for simple actions
+
+MOCK_OUTPUT_1 = {
+    'execution_id': 'idfoo3',
+    'timestamp': '1505732598',
+    'output_type': 'stdout',
+    'data': 'line 1\n'
+}
+
+MOCK_OUTPUT_2 = {
+    'execution_id': 'idfoo3',
+    'timestamp': '1505732598',
+    'output_type': 'stderr',
+    'data': 'line 2\n'
+}
+
+
+class ActionExecutionTailCommandTestCase(BaseCLITestCase):
```

Comment:
I'm not too happy with the tests because it results in a lot of mocking and that's fragile in case the the format changes.

We have similar problem in many other places so we should figure out a better way to test it. Another option would be to add some robot tests, perhaps in addition to those unit tests.